packages/common/src/viemActions.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+  WriteContractParameters,
+  Transport,
+  Chain,
+  Account,
+  WalletActions,
+  WalletClient,
+  WriteContractReturnType,
+} from "viem";
+import { getAction } from "viem/utils";
+import { writeContract } from "viem/actions";
+import { writeContract as mud_writeContract } from "./writeContract";
+import { sendTransaction as mud_sendTransaction } from "./sendTransaction";
+import { type ContractWrite } from "./getContract";
+
+export const sessionActions = <TChain extends Chain, TAccount extends Account>(
+  client: WalletClient<Transport, TChain, TAccount>
+): Pick<WalletActions<TChain, TAccount>, "writeContract" | "sendTransaction"> => {
+  return {
+    // Applies to: `client.writeContract`, `getContract(client, ...).write`
+    writeContract: (args) => mud_writeContract(client, args),
+    // Applies to: `client.sendTransaction`
+    sendTransaction: (args) => mud_sendTransaction(client, args),
+  };
+};
+
+export const setupObserverActions = <TChain extends Chain, TAccount extends Account>(
```

Comment:
maybe `writeObserverActions`?

Reply:
I like `writeObserverActions`, but I feel we want something like `setup` for distinguishability.

Given this naming convention:
```ts
client.extend(walletActions).extend(transactionQueueActions)
```

This feels natural:
```ts
client.extend(writeObserverActions)
```

But this looks different:
```ts
client.extend(writeObserverActions(onWrite))
```

Reply:
I see what you mean now. I'd recommend we use the pattern that [viem does](https://viem.sh/op-stack#3-consume-op-stack-actions) (and generally mirrors "middleware" approaches of libraries [like express.js](https://expressjs.com/en/resources/middleware.html)) by making them all functions, so we can eventually support passing in options to each as needed.

So they'd become

```
client
  .extend(transactionQueueActions())
  .extend(writeObserverActions())
```

I could also see moving this into an `actions` dir and naming them each `transactionQueue` and `writeObserver` (to avoid/clean up the "actions" prefix). But I don't feel as strongly there.

Reply:
> making them all functions, so we can eventually support passing in options to each as needed

I agree with this approach for future extendability. I'll employ the `writeObserver({ onWrite })` structure here.

I didn't realize viem already have this function style actions.